google/cloud/spanner_dbapi/cursor.py

```diff
@@ -293,10 +297,16 @@ def execute(self, sql, args=None):
                     args or None,
                 )
         except (AlreadyExists, FailedPrecondition, OutOfRange) as e:
+            self.close()
+            self.connection.close()
```

Comment:
1. I am closing the connection as the state of the connection doesn't remain proper (for example _spanner_transaction_started flag) and if the Connection is reused it would lead to issues.
2. I wanted to do that but couldn't found a way. Let me try to look for more

Reply:
Updated the code to not close the connection and update the flag if needed in finally block